### ISSUE TITLE
Help with CentOS and gnu()

### ISSUE THREAD

**Issue by mathomp4** [NVIDIA/hpc-container-maker#228]
All,

I'm trying to learn how to use Docker, etc., and figured I'd lean on NVIDIA's experts in creating non-trivial build images/containers (I still am not fluent with the lingo) for things like an GCC 9/Open MPI 4.0.2 setup.

So, I created a recipe cribbed off of the hpcbase-gnu-openmpi one but using CentOS 8 to try and get close to a Dockerfile a more knowledgeable coworker created by hand. To wit:
```python
Stage0 += comment(__doc__, reformat=False)

Stage0 += baseimage(image="centos:8")

# Python
Stage0 += python()

# GNU compilers
compiler = gnu(version='9')
#compiler = gnu(extra_repository=True, version='9')
#compiler = gnu(extra_repository=False, version='9')
Stage0 += compiler

# OpenMPI
Stage0 += openmpi(cuda=False, infiniband=False, 
        version='4.0.2', toolchain=compiler.toolchain)
```

However, when I try to build this:
```
$ docker build -t fortran/gcc9-openmpi402:v1.0.0 -f Dockerfile  .
Sending build context to Docker daemon  6.656kB
Step 1/7 : FROM centos:8
 ---> 470671670cac
Step 2/7 : RUN yum install -y         python2         python3 &&     rm -rf /var/cache/yum/*
 ---> Using cache
 ---> 02444ced632f
Step 3/7 : RUN yum install -y centos-release-scl &&     yum install -y         devtoolset-9-gcc         devtoolset-9-gcc-c++         devtoolset-9-gcc-gfortran &&     rm -rf /var/cache/yum/*
 ---> Running in ef980a85cdcd
Last metadata expiration check: 0:02:39 ago on Wed Feb 12 14:46:59 2020.
No match for argument: centos-release-scl
Error: Unable to find a match: centos-release-scl
The command '/bin/sh -c yum install -y centos-release-scl &&     yum install -y         devtoolset-9-gcc         devtoolset-9-gcc-c++         devtoolset-9-gcc-gfortran &&     rm -rf /var/cache/yum/*' returned a non-zero code: 1
```
I've tried all the variants seen above:
```python
compiler = gnu(version='9')
compiler = gnu(extra_repository=True, version='9')
compiler = gnu(extra_repository=False, version='9')
```
but none of them seem to work (which seems to track with the "if you set `version` you get extra in the docs).

I suppose the main question is: what have I done wrong? I'm sure it's simple, but I'm a bit lost right now.

**Comment by mathomp4** [comment]
I just did some tests and it looks like:
```python
Stage0 += baseimage(image="ubuntu")
...
compiler = gnu(extra_repository=True, version='9')
```
will build.

**Comment by samcmill** [comment]
@mathomp4, glad to hear you got the Ubuntu version working.  

It looks like there's an issue with the centos:8 handling - some of the names changed between CentOS 7 and 8:

This works for CentOS 8:

```
# yum install centos-release-stream
# yum install gcc-toolset-9-gcc gcc-toolset-9-gcc-c++ gcc-toolset-9-gcc-gfortran
```

I'll add some special handling for the centos:8 case to reflect what's above.

**Comment by mathomp4** [comment]
@samcmill Thanks! I'm sure I'll be bothering you a lot more soon. I have a set of Base libraries I build for a climate model I work on and I'm trying to build them via Docker. I'm hoping to use `generic_build()` to do so.

**Comment by samcmill** [comment]
The CentOS 8 compiler issue is resolved by #229
